artpaint/paintwindow/PaintWindow.cpp

```diff
@@ -316,15 +291,27 @@ PaintWindow::PaintWindow(BRect frame, const char* name, uint32 views,
 		PostMessage(&msg, this);
 	}
 
+	BGroupLayout* layout = BLayoutBuilder::Group<>(this, B_VERTICAL, 0)
+//		.Add(fMenubar)
+		.AddGroup(B_HORIZONTAL, 0)
+			.AddGroup(B_VERTICAL, 0)
+				.Add(fBackground)
+//				.Add(fStatusView, 0)
+				.End()
+			.Add(fVerticalScrollbar)
+			.End()
+		.Add(fHorizontalScrollbar);
+
+	if (fMenubar) layout->AddView(0, fMenubar);
+
+	if (fStatusView)
+		((BGroupLayout*) ((BGroupLayout*) layout->ItemAt(fMenubar != NULL))->ItemAt(0))->AddView(fStatusView, 0);
```

Comment:
This looks a bit ugly. But it can be improved.

- Handle the status view before the menu bar. It will save you the "fMenuBar != NULL" trick.
- It may be easier to do it in a different way:

```
BGroupLayout * layout = new BGroupLayout(B_VERTICAL, 0);
layout->Add(fBackground);
if (fStatusView) layout->Add(fStatusView, 0);
```

And later on in your builder:

```
.Add(layout) // replaces lines 297-300
```